### PR TITLE
buildconf, Makefile: standardize and be more consistent

### PR DESCRIPTION
**Pull request by Somasis**
Same changes as the ones sent to fskit and libpstat.

### CODE REVIEW

**Comment by jcnelson** [comment]
Looks good; merged!

**Comment by fbt** [comment]
Interesting. I have a problem building vdevfs after this commit:

    In file included from /usr/include/fskit/fskit.h:25:0,
                 from fs.h:32,
                 from main.h:26,
                 from main.c:22:
    /usr/include/fskit/common.h:63:20: fatal error: iostream: No such file or directory

But if I change CC to CXX here, it works:

    $(BUILD_VDEVFS)/%.o : %.c
        @mkdir -p "$(shell dirname "$@")"
        $(CC) $(CFLAGS) $(INC) $(DEFS) -o "$@" -c "$<"

Basically I've googled similar problems and tried something blindly, and it worked.
Judging by what other people have encoutered, your main.c contains C++ code, which would obviously cause problems if we try to build it with gcc.

**Comment by fbt** [comment]
Nevermind my previous comment, I've created a separate issue, as it is not directly related to this specific commit. The makefile is actually fine.
